src/aws-api-mcp-server/awslabs/aws_api_mcp_server/core/agent_scripts/manager.py

```diff
@@ -0,0 +1,71 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import frontmatter
+from .models import Script
+from pathlib import Path
+from typing import Optional
+
+
+class AgentScriptsManager:
+    """Script manager for AWS API MCP."""
+
+    _instance: Optional['AgentScriptsManager'] = None
+
+    def __new__(cls) -> 'AgentScriptsManager':
```

Comment:
Is this truly needed? We can instantiate this at module level once, which effectively acts as a single ton due to the nature on how imports work.